modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -721,11 +720,10 @@ uint32_t PngDecoder::read_chunk(Chunk& chunk)
         if (size != 8 + 26 + 4)
             return 0;
     } else if (id == id_bKGD) {
-        // 8=HDR+size, ??=size of bKGD chunk, 4=CRC
+        // 8=HDR+size, 6=size of bKGD chunk, 4=CRC
         // The spec is actually more complex:
         // http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.bKGD
-        // TODO: we only check that 4 bytes can be read from &chunk.p[8]. Fix.
-        if (size < 8 + 4)
+        if (size < 8 + 6)
```

Comment:
Actually, let's follow the spec: `size != 8 + 1 && size != 8 + 2 && size != 8 + 6`